dallinger/prolific.py

```diff
@@ -281,3 +281,91 @@ def _req(self, method: str, endpoint: str, **kw) -> dict:
             raise ProlificServiceException(json.dumps(error))
 
         return parsed
+
+
+class DevProlificService(ProlificService):
+    """Wrapper that mocks the Prolific REST API and instead of making requests it writes to the log."""
+
+    def __init__(self, api_token: str, api_version: str, referer_header: str):
+        super().__init__(api_token, api_version, referer_header)
+
+    def approve_participant_session(self, session_id: str) -> dict:
```

Comment:
Not obvious to me that `approve_participant_session` should be overridden. Wouldn't it be better to leave this untouched, and instead patch `get_participant_session`?

Reply:
:+1: